### PR TITLE
fix: Pinning dependency versions in tests

### PR DESCRIPTION
**Pull request by bitsandfoxes**
Based on the most frustrating of all development experiences. New versions of packages were released but could not be downloaded. This resulted in the greatest of confusions and broke my local environment. (not for the first time)
I don't see a good reason, especially for tests, why versions of dependencies cannot be pinned.

I'd rather have them pinned and add a bumping mechanism instead of getting surprised.

#skip-changelog

### CODE REVIEW

**Comment by SeanFeldman** [comment]
> I'd rather have them pinned and add a bumping mechanism instead of getting surprised.

Would Dependabot or Renovate help? In general, having the latest of the same major version is better for testing to identify issues earlier.

**Comment by bitsandfoxes** [comment]
> > I'd rather have them pinned and add a bumping mechanism instead of getting surprised.
> 
> Would Dependabot or Renovate help? In general, having the latest of the same major version is better for testing to identify issues earlier.

Super valid point!

**Comment by SeanFeldman** [comment]
> Super valid point!

With Dependabot already [in place](https://github.com/getsentry/sentry-dotnet/pulls?q=is%3Apr+is%3Aclosed+dependabot), what else I do not see that would require pinning for test projects?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -5,6 +5,12 @@ updates:
   schedule:
     interval: monthly
   open-pull-requests-limit: 2
+
+- package-ecosystem: nuget
```
Comment (bitsandfoxes) [inline .github/dependabot.yml:9]: @jamescrosswell I'm not too familiar with dependabot but that should do the trick?

Comment (SeanFeldman) [reply .github/dependabot.yml:9]: @bitsandfoxes yes. I'm unsure about the number of PRs that can be opened. Two sounds very constraining for a weekly period.

Comment (bruno-garcia) [reply .github/dependabot.yml:9]: On the other hand not very busy/distracting, we can adjust as we go

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**